main.go

```diff
@@ -300,45 +323,99 @@ func executeCLI(cmd *cobra.Command, src *source, w io.Writer) error {
 		return fmt.Errorf("unable to create renderer: %w", err)
 	}
 
-	content := string(b)
-	ext := filepath.Ext(src.URL)
-	if isCode {
-		content = utils.WrapCodeBlock(string(b), ext)
-	}
+	// Stream with incremental rendering using context
+	renderFunc := func(content []byte) []byte {
+		// Detect truly pathological patterns that will hang glamour
+		contentLen := len(content)
 
-	out, err := r.Render(content)
-	if err != nil {
-		return fmt.Errorf("unable to render markdown: %w", err)
+		// Check for excessive unclosed brackets that cause O(n²) parsing
```

Comment:
I'm not sure if this is worth doing; I'll do some investigation here and vet this part a bit more before dropping DRAFT.

Reply:
This no longer exists and the line limit works better anyway